src/simulation/controller.rs

```diff
@@ -78,7 +82,7 @@ pub fn run_mpi() {
     info!("Process #{} finishing.", world.rank());
 }
 
-fn execute_partition<C: NetCommunicator>(comm: C, config: Arc<Config>) {
+fn execute_partition<C: SimCommunicator + 'static>(comm: C, config: Arc<Config>) {
```

Comment:
Otherwise we gat a compile error (E0310). It's not very helpful, but this issue is: https://github.com/rust-lang/rust/issues/103849 
The reason is the following: In replanner.rs:97 we construct a Box<dyn Router>, which is shorthand for Box<dyn Router + 'static). The implementing TravelTimeCollectingAltRouter has the generic C: SimCommunicator. Thus, it must also be 'static. 
When I tried it with a DummyRouter that has no generic, I didn't need 'static. I think I could have also annotated lifetimes to the Router and it's parents (as proposed here https://stackoverflow.com/questions/29740488/parameter-type-may-not-live-long-enough), but for the SimCommunicator it makes sense to me to annotate it with 'static.